extension/oauth2clientauthextension/extension.go

```diff
@@ -38,6 +39,22 @@ type ClientCredentialsAuthenticator struct {
 // ClientCredentialsAuthenticator implements ClientAuthenticator
 var _ configauth.ClientAuthenticator = (*ClientCredentialsAuthenticator)(nil)
 
+type errorWrappingTokenSource struct {
+	ts     oauth2.TokenSource
+	config *clientcredentials.Config
+}
+
+// errorWrappingTokenSource implements TokenSource
+var _ oauth2.TokenSource = (*errorWrappingTokenSource)(nil)
+
+// FailedToGetSecurityToken indicates a problem communicating with OAuth2 server.
+// We support Unwrap() instead of using `%w` so that we can customize the error message
+// to include both the wrapped error and information from the configuration.
+type FailedToGetSecurityTokenError struct {
```

Comment:
Can we make this private - i don't see why it needs to be exported

Reply:
Please see my comment above, don't think we need this struct

Reply:
personally, i think there is no need of the second approach as our primary concern is to pop the new error with enough data to caller. At the moment In our exporters  (callers) we treat all  the errors in same fashion so we can do with approach 1